cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionMetadata.java

```diff
@@ -27,11 +27,13 @@
  */
 public class PartitionMetadata implements Iterable<Map.Entry<String, String>> {
   private final Map<String, String> metadata;
-  private final Long creationTime;
+  private final long creationTime;
+  private final long lastUpdatedTime;
 
-  public PartitionMetadata(Map<String, String> metadata, long creationTime) {
+  public PartitionMetadata(Map<String, String> metadata, long creationTime, long lastUpdatedTime) {
```

Comment:
I prefer `lastModifiedTime` (similar to file attributed `lastModifiedTime` in Java).